include/qengine_opencl.hpp

```diff
@@ -35,33 +32,49 @@ typedef std::shared_ptr<QEngineOCL> QEngineOCLPtr;
 /** OpenCL enhanced QEngineCPU implementation. */
 class QEngineOCL : public QEngineCPU {
 protected:
+    int deviceID;
     OCLEngine* clObj;
+    DeviceContextPtr device_context;
     cl::CommandQueue queue;
-    cl::Buffer stateBuffer;
+    cl::Context context;
+    BufferPtr stateBuffer;
     cl::Buffer cmplxBuffer;
     cl::Buffer ulongBuffer;
     cl::Buffer nrmBuffer;
     cl::Buffer maxBuffer;
 
 public:
-    QEngineOCL(bitLenInt qBitCount, bitCapInt initState, std::shared_ptr<std::default_random_engine> rgp = nullptr)
-        : QEngineCPU(qBitCount, initState, rgp)
+    QEngineOCL(bitLenInt qBitCount, bitCapInt initState, std::shared_ptr<std::default_random_engine> rgp = nullptr,
+        int devID = -1, bool partialInit = false)
+        : QEngineCPU(qBitCount, initState, rgp, complex(-999.0, -999.0), partialInit)
+    {
+        InitOCL(devID);
+    }
+
+    QEngineOCL(QEngineOCLPtr toCopy)
+        : QEngineCPU(toCopy->GetQubitCount(), 0, toCopy->rand_generator, complex(-999.0, -999.0), false)
+    {
+        doNormalize = toCopy->doNormalize;
```

Comment:
Repeat comment about initializing in parameter list.

Reply:
Fixed.